### PR TITLE
Add unit test coverage for init container phases

### PR DESCRIPTION
**Pull request by ehashman**
#### What type of PR is this?

/kind cleanup
/sig node
/priority important-soon

because there are a few bugfixes that would benefit from this coverage

#### What this PR does / why we need it:

Adds unit test cases for the spread of init container states.

#### Which issue(s) this PR fixes:

Despite there being a large amount of phase logic for init containers, there are currently no unit tests. I am trying to fix a phase bug and wanted to ensure I didn't cause any regressions, and Clayton's refactor may also benefit from this.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/kubelet_pods_test.go [hunk 1]
```diff
@@ -2083,6 +2302,10 @@ func TestPodPhaseWithRestartOnFailure(t *testing.T) {
 	}
 }
 
+// No special init-specific logic for this, see RestartAlways case
+// func TestPodPhaseWithRestartOnFailureInitContainers(t *testing.T) {
+// }
```
Comment (ehashman) [inline pkg/kubelet/kubelet_pods_test.go:2307]: I could copy-paste the test cases for RestartAlways for the RestartOnFailure cases if someone thinks there's any value... put this comment here as an explanation for why I didn't.

Comment (smarterclayton) [reply pkg/kubelet/kubelet_pods_test.go:2307]: I'm fine with that.

**Comment by ehashman** [comment]
flake #99979

I thought that was fixed :/

/test pull-kubernetes-e2e-kind-ipv6

**Comment by matthyx** [comment]
> flake #99979
> 
> I thought that was fixed :/
> 
> /test pull-kubernetes-e2e-kind-ipv6

@ehashman your branch didn't include my fix here:
https://github.com/ehashman/kubernetes/blob/init-container-coverage/pkg/kubelet/prober/prober_manager.go#L245

**Comment by ehashman** [comment]
That shouldn't matter, PRs will get merged into master before testing.

**Comment by matthyx** [comment]
> That shouldn't matter, PRs will get merged into master before testing.

I think it does both... first the merge on master at the revision you have branched from, and then on master-latest where your commit will land.
@BenTheElder could you confirm?

**Pull request merged**